build/helper/codegen_helper.py

```diff
@@ -426,7 +435,10 @@ def _get_ctype_variable_definition_snippet_for_buffers(parameter, parameters, iv
     definition = None
 
     if parameter['numpy'] is True and use_numpy_array is True:
-        definition = '_get_ctypes_pointer_for_buffer(value={})  # case B510'.format(parameter['python_name'])
+        if parameter['complex_type'] == 'none':
+            definition = '_get_ctypes_pointer_for_buffer(value={})  # case B510'.format(parameter['python_name'])
+        else:
+            definition = '_get_ctypes_pointer_for_buffer(value={}, library_type={}.{})  # case B510'.format(parameter['python_name'], module_name, parameter['ctypes_type'])
```

Comment:
@marcoskirsch do you know where these BXXX sequences come from. Is this correct usage? It feels to me like it should be separate from B510.